src/runtime/write_debug_image.cpp

```diff
@@ -101,14 +101,65 @@ WEAK bool has_tiff_extension(const char *filename) {
     return *f == '\0';
 }
 
+int32_t get_data_index(int32_t dim0, int32_t dim1, int32_t dim2, int32_t dim3,
+                       const int32_t *strides) {
+    int32_t index = dim0*strides[0] + dim1*strides[1] + dim2*strides[2] + dim3*strides[3];
```

Comment:
Probably don't come up for internal allocations, but the host pointer is the address of the min coordinate, not the address of the origin, so this needs to be (dim0 - min[0])*stride[0] + (dim1 - min[1])*stride[1] + ....

Probably should just pass the buffer_t through to this function.